pkg/asset/agent/image/iso.go

```diff
@@ -0,0 +1,80 @@
+package image
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	"github.com/openshift/installer/pkg/agent/isosource"
+	"github.com/openshift/installer/pkg/asset"
+)
+
+const (
+	agentISOFilename = "output/fleeting.iso"
+)
+
+// ISO is an asset that generates the bootable image used to install clusters.
+type ISO struct {
```

Comment:
I'd suggest AgentImage as the name, since this may not always be an ISO forever.